enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/assets/RunInClusterAction.java

```diff
@@ -0,0 +1,281 @@
+        try (KubernetesClient client = new KubernetesClientBuilder().withConfig(config).build();) {
+            consumer.accept(client);
+        } catch (Exception e) {
+            Exceptions.printStackTrace(e);
+        }
+    }
+
+    private Config prepareConfig(String content, ClusterItem cluster) {
+        String token = getBearerToken(cluster);
+        LoadSettings settings = LoadSettings.builder().build();
+        Load load = new Load(settings);
+        Map<String, Object> data = (Map<String, Object>) load.loadFromString(content);
+
+        List<Map<String, Object>> users = (List<Map<String, Object>>) data.get("users"); //NOI18N
+
+        Map exec = null;
+        for (Map<String, Object> userEntry : users) {
+            Map<String, Object> user = (Map<String, Object>) userEntry.get("user"); //NOI18N
+            if (user != null && user.containsKey("exec")) {
+                exec = (Map) user.remove("exec");
+            }
+        }
+        String clusterId = null;
+        if (exec != null && "oci".equals(exec.get("command"))) { //NOI18N
+            List commandArgs = (List) exec.get("args"); //NOI18N
+            boolean clusterIdNext = false;
+            for (Object arg : commandArgs) {
+                if ("--cluster-id".equals(arg)) {
+                    clusterIdNext = true;
+                    continue;
+                }
+                if (clusterIdNext) {
+                    clusterId = (String) arg;
+                    break;
+                }
+
+            }
+        }
+        if (!cluster.getKey().getValue().equals(clusterId)) {
+            throw new RuntimeException("Failed to read cluster config"); //NOI18N
+        }
+        Dump dump = new Dump(DumpSettings.builder().build());
+        String noExec = dump.dumpToString(data);
+        Config config = Config.fromKubeconfig(noExec);
+        config.setOauthToken(token);
+        return config;
+    }
+
+    private String getBearerToken(ClusterItem cluster) {
+        try {
+            OCIProfile profile = OCIManager.getDefault().getActiveProfile(cluster);
+            Region region = Region.fromRegionCodeOrId(cluster.getRegionCode());
+            URI uri = new URI(String.format(
+                    "https://containerengine.%s.oraclecloud.com/cluster_request/%s", //NOI18N
+                    region.getRegionId(), cluster.getKey().getValue()
+            ));
+
+            RequestSigningFilter requestSigningFilter
+                    = RequestSigningFilter.fromAuthProvider(profile.getAuthenticationProvider());
+
+            HttpClient client = JerseyHttpProvider.getInstance().newBuilder()
+                    .registerRequestInterceptor(Priorities.AUTHENTICATION, requestSigningFilter)
+                    .baseUri(uri).build();
+
+            HttpRequest request = client.createRequest(Method.GET);
+
+            request.execute().toCompletableFuture();
+            List<String> tokenUrlList = request.headers().get("authorization"); //NOI18N
```

Comment:
Can be `autorization` sometimes missing from the headers ?

Reply:
This is mandatory for all requests and was present every time while testing.